game.exs

```diff
@@ -1,7 +1,58 @@
 defmodule GuessingGame do
 
-  def hi() do
-    IO.puts("hello world")
+  def compare(n, t) do
+    # Best Practice ?
+
+    # a)
+    # case n do
+    #   m when m < t -> :low
+    #   m when m == t -> :equal
+    #   m when m > t -> :high
+    # end
+
+    # b)
+    cond do
+      n < t -> :low
+      n == t -> :equal
+      n > t -> :high
+    end
+  end
+
+  def print_state(state) do
+    case state do
```

Comment:
good use of case, it also would work to simple define `print_state` three times with a different pattern for `state` in the function argument.

However, I think this is good as a case statement.